shell/src/main/java/alluxio/cli/fsadmin/metrics/ClearCommand.java

```diff
@@ -0,0 +1,259 @@
+        List<WorkerNetAddress> addressList = store.getEligibleWorkers().stream()
+            .map(BlockWorkerInfo::getNetAddress).collect(Collectors.toList());
+        if (!clearWorkers(addressList, context)) {
+          return -1;
+        }
+      }
+    }
+
+    if (options.length == 0 || cl.hasOption(MASTER_OPTION_NAME)) {
+      // Clear worker metrics before master metrics since worker metrics report
+      // may be flaky during metrics clearance
+      try {
+        mMetricsClient.clearMetrics();
+        System.out.printf("Successfully cleared metrics of Alluxio leading master.%n");
+      } catch (Exception e) {
+        System.out.println("Fatal error: " + e);
+        return -1;
+      }
+    }
+    return 0;
+  }
+
+  /**
+   * Clears the metrics of a list of workers.
+   *
+   * @param workers the workers to clear metrics of
+   * @param context FileSystemContext
+   * @return true if clear succeed, false otherwise
+   */
+  private boolean clearWorkers(List<WorkerNetAddress> workers,
+      FileSystemContext context) throws IOException {
+    int workerNum = workers.size();
+    if (workerNum == 0) {
+      System.out.println("No worker metrics to clear.");
+      return true;
+    } else if (workerNum == 1) {
+      clearWorkerMetrics(workers.get(0), context);
+    } else {
+      List<Future<Void>> futures = new ArrayList<>();
+      int parallelism = Math.min(workerNum, DEFAULT_PARALLELISM);
+      ExecutorService service = Executors.newFixedThreadPool(parallelism,
+          ThreadFactoryUtils.build("metrics-clear-cli-%d", true));
+      for (WorkerNetAddress worker : workers) {
+        futures.add(service.submit(new ClearCallable(worker, context)));
+      }
+      try {
+        for (Future<Void> future : futures) {
+          future.get();
+        }
+      } catch (ExecutionException e) {
+        System.out.println("Fatal error: " + e);
+        return false;
+      } catch (InterruptedException e) {
+        System.out.println("Metrics clearance interrupted, exiting.");
+        return false;
+      } finally {
+        service.shutdownNow();
+      }
+    }
+    return true;
+  }
+
+  /**
+   * Thread that clears the metrics of a specific worker.
+   */
+  private class ClearCallable implements Callable<Void> {
+    private final WorkerNetAddress mWorker;
+    private final FileSystemContext mContext;
+
+    ClearCallable(WorkerNetAddress worker, FileSystemContext context) {
+      mWorker = worker;
+      mContext = context;
+    }
+
+    @Override
+    public Void call() throws Exception {
+      clearWorkerMetrics(mWorker, mContext);
+      return null;
+    }
+  }
+
+  /**
+   * Clears the worker metrics.
+   *
+   * @param worker the worker to clear metrics of
+   * @param context the file system context
+   */
+  private void clearWorkerMetrics(WorkerNetAddress worker,
+      FileSystemContext context) throws IOException {
+    BlockWorkerClient blockWorkerClient = context.acquireBlockWorkerClient(worker);
```

Comment:
can we use try-with-resources here?

Reply:
```
        BlockWorkerClient blockWorker = mContext.acquireBlockWorkerClient(worker);
        try {
          blockWorker.asyncCache(request);
          mLastBlockIdCached = blockId;
        } finally {
          mContext.releaseBlockWorkerClient(worker, blockWorker);
        }
```
In the client code, we always use `mContext.releaseBlockWorkerClient` instead of directly close the `blockWorkerClient`. This is for resources reuse and reduce the effort of creating/closing frequently-used resources